app/scripts/lib/account-tracker.js

```diff
@@ -158,90 +300,156 @@ export default class AccountTracker {
+      });
+    });
     // save accounts state
-    this.store.updateState({ accounts });
+    this.store.updateState({ accounts, accountsByChainId });
   }
 
   /**
    * Removes all addresses and associated balances
    */
-
   clearAccounts() {
-    this.store.updateState({ accounts: {} });
+    this.store.updateState({
+      accounts: {},
+      accountsByChainId: {
+        [this.getCurrentChainId()]: {},
+      },
+    });
   }
 
   /**
-   * Given a block, updates this AccountTracker's currentBlockGasLimit, and then updates each local account's balance
-   * via EthQuery
+   * Given a block, updates this AccountTracker's currentBlockGasLimit and currentBlockGasLimitByChainId and then updates
+   * each local account's balance via EthQuery
    *
    * @private
    * @param {number} blockNumber - the block number to update to.
    * @fires 'block' The updated state, if all account updates are successful
    */
   async _updateForBlock(blockNumber) {
-    this._currentBlockNumber = blockNumber;
+    await this._updateForBlockByNetworkClientId(null, blockNumber);
+  }
+
+  /**
+   * Given a block, updates this AccountTracker's currentBlockGasLimitByChainId, and then updates each local account's balance
+   * via EthQuery
+   *
+   * @private
+   * @param {string} networkClientId - optional network client ID to use instead of the globally selected network.
+   * @param {number} blockNumber - the block number to update to.
+   * @fires 'block' The updated state, if all account updates are successful
+   */
+  async _updateForBlockByNetworkClientId(networkClientId, blockNumber) {
+    const { chainId, provider } =
+      this.#getCorrectNetworkClient(networkClientId);
+    this._currentBlockNumberByChainId[chainId] = blockNumber;
 
     // block gasLimit polling shouldn't be in account-tracker shouldn't be here...
-    const currentBlock = await this._query.getBlockByNumber(blockNumber, false);
+    const currentBlock = await pify(new EthQuery(provider)).getBlockByNumber(
+      blockNumber,
+      false,
+    );
     if (!currentBlock) {
       return;
     }
     const currentBlockGasLimit = currentBlock.gasLimit;
-    this.store.updateState({ currentBlockGasLimit });
+    const { currentBlockGasLimitByChainId } = this.store.getState();
+    currentBlockGasLimitByChainId[chainId] = currentBlockGasLimit;
+    this.store.updateState({ currentBlockGasLimitByChainId });
+    if (chainId === this.getCurrentChainId()) {
+      this.store.updateState({
+        currentBlockGasLimit,
+      });
+    }
 
     try {
-      await this._updateAccounts();
+      await this.updateAccounts(networkClientId);
     } catch (err) {
       log.error(err);
     }
   }
 
+  /**
+   * Updates accounts for the globally selected network
+   * and all networks that are currently being polled.
+   *
+   * @returns {Promise} after all account balances updated
+   */
+  async updateAccountsAllActiveNetworks() {
+    await this.updateAccounts();
+    await Promise.all(
```

Comment:
Wondering if we might want to use `Promise.allSettled` here instead since it won't reject for all if one request rejects. Alternatively we could wrap each promise in the array in a try/catch. wdyt @mcmire @shanejonas ?